### PR TITLE
Remove duplication of GRPC configuration

### LINKED ISSUE #5446: Remove duplication of GRPC configuration (client and server)

### PR DESCRIPTION
**Pull request by aledbf**
fixes #5446

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#5446*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/blobserve/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/blobserve/OWNERS)**
- **[components/common-go/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/common-go/OWNERS)**
- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- **[components/ee/kedge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/kedge/OWNERS)**
- **[components/ee/ws-scheduler/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/ws-scheduler/OWNERS)**
- **[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)**
- **[components/registry-facade/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/registry-facade/OWNERS)**
- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**
- **[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)**
- **[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/common-go/grpc/grpc.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	grpc_opentracing "github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing"
+	"github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/keepalive"
+)
+
+// maxMsgSize use 16MB as the default message size limit.
+// grpc library default is 4MB
+const maxMsgSize = 1024 * 1024 * 16
```
Comment (aledbf) [inline components/common-go/grpc/grpc.go:20]: The main reason to adjust this value is related to some observed metrics that could indicate the current value is low.
`grpc_server_handled_total{grpc_code="ResourceExhausted",grpc_method="StartWorkspace",grpc_service="wsman.WorkspaceManager",grpc_type="unary"} 8
`

**Comment by csweichel** [comment]
Love the approach. Two comments:
1. In a follow up we could harmonise the TLS handling across the components, too. Right now we have all the TLS code - esp. on the client side - strewn across the components.
2. What about the TS components? It would probably make sense to introduce a similar mechanism in gitpod-protocol, no?

**Comment by aledbf** [comment]
> Love the approach. Two comments:
> 
> 1. In a follow up we could harmonise the TLS handling across the components, too. Right now we have all the TLS code - esp. on the client side - strewn across the components.

Yes, I saw that ;)

> 2. What about the TS components? It would probably make sense to introduce a similar mechanism in gitpod-protocol, no?

Already WIP https://github.com/gitpod-io/gitpod/pull/5449

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 10fd1fcd8c86303410072362baed566bc52c16fc</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5447#issuecomment-910410187" title="LGTM">csweichel</a>*

Associated issue: *#5446*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]
- ~~[components/blobserve/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/blobserve/OWNERS)~~ [csweichel]
- ~~[components/common-go/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/common-go/OWNERS)~~ [csweichel]
- ~~[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)~~ [csweichel]
- ~~[components/ee/kedge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/kedge/OWNERS)~~ [csweichel]
- ~~[components/ee/ws-scheduler/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/ws-scheduler/OWNERS)~~ [csweichel]
- ~~[components/image-builder/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder/OWNERS)~~ [csweichel]
- ~~[components/registry-facade/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/registry-facade/OWNERS)~~ [csweichel]
- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [csweichel]
- ~~[components/ws-manager/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager/OWNERS)~~ [csweichel]
- ~~[components/ws-proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-proxy/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**